### DISCUSSION TITLE
Gunicorn + Gevent Websocket

### DISCUSSION THREAD

**Discussion by rajan-code** [miguelgrinberg/Flask-SocketIO#2028] [General]
Let's say I run the server with 
`gunicorn -k geventwebsocket.gunicorn.workers.GeventWebSocketWorker -w 1 app:app`
How many concurrent requests can this handle? 
Can I safely modify global variables without a mutex?

**Comment by miguelgrinberg** [comment]
How many request can be handled will depend on the resources available on your server and how much of them each request uses. This cannot be answered generically. Normally you would measure what volume of requests your application can handle by running tests on a server that is similar to your production server.

Modifying global variables is easier when using gevent than when using threads, but it is also something that will largely depend on how your application is designed. So in general I'd say that no, it is not always possible. But you can design your app to allow this if you are careful.
